lib/guard.js

```diff
@@ -131,18 +134,32 @@ const MetricsGuard = class MetricsGuard extends stream.Transform {
 			this.registry.set(metric.name, entry);
 		}
 
-		// push all label permutations on the metric into the
-		// Set for the metric in the registry. the size of
-		// this Set is compared with the threshold
-		metric.labels.forEach((label) => {
-			entry.add(`${label.name}${SEPARATOR}${label.value}`);
-		});
-
-		if (metric.source === this.id) {
-			this.emit("drop", metric);
-			next(null);
-			return;
-		}
+        let hasInvalidLabelValue = [];
+        // push all label permutations on the metric into the
+        // Set for the metric in the registry. the size of
+        // this Set is compared with the threshold
+        metric.labels.forEach((label) => {
+            if (label && (
+                (label.name === null || typeof label.name === 'undefined')
+                || (label.value === null || typeof label.value === 'undefined')
```

Comment:
The `name` is one thing, but can the value not be null or undefined? 🤔

Reply:
that is what we want to warn against, as you will get the error about arguments and labels not matching if value if null of undefined. right?

Reply:
For labels yes, but I think a label's value can be falsy and not fail. I can write a test.

Reply:
I added this test to `histogramTest.js` in `prom-client` and it passes:
```js
it('should not crash on falsy values', () => {
  let i = new Histogram({
    name: 'histo',
    help: 'help',
    labelNames: ['code'],
  });
  i = i.labels(undefined);
  expect(true).toEqual(true);
});
```

This one fails because we get two (falsy) values for the one configured label name:

```js
```js
it('should not crash on falsy values', () => {
  let i = new Histogram({
    name: 'histo',
    help: 'help',
    labelNames: ['code'],
  });
  i = i.labels(undefined, undefined);
  expect(true).toEqual(true);
});
```
```

Reply:
👍 I'll add a test for this here as well.